### PR TITLE
commit tensorflow builds to pagure repository

### PR DESCRIPTION
**Pull request by harshad16**
The Tensorflow builds are to be pushed to Pagure as a git commit.
- In the build configs added parameters:
  - PAGURE_COMMIT_REPO:<pagure/tensorflow>, PAGURE_COMMIT_WHL:<0/1>
  - changed GIT_DEST_REPO <-> GIT_RELEASE_REPO:<AICoE/wheels>
- Added functionality if PAGURE_COMMIT_WHL is set 1, then the files are pushed to Pagure Tensorflow repository else releases the wheel files as a git release in AICoE repository.
- Added secret config and parameter to pass ssh key for Pagure repository connection.
  - PAGURE_SSH_PRIVATE_KEY: Parameter to pass private ssh key for pagure authentication. 

Updated the naming convention of the wheel file storage repository.
At this moment the wheel files are pushed with `hnalla@redhat.com` git configuration to pagure/tensorflow. Will be modified as per directions :) 
@goern  @sub-mod

### CODE REVIEW

File: tensorflow-build-job.json [hunk 1]
```diff
@@ -383,10 +427,28 @@
...
+        {
+            "name": "PAGURE_COMMIT_REPO",
+            "description": "PAGURE_COMMIT_REPO",
+            "value": "ssh://git@pagure.io/tensorflow.git",
+            "required": true
+        },
+        {
+            "name": "PAGURE_SSH_PRIVATE_KEY",
+            "description": "SSH private key to authenticate the public key set in GitHub.",
+            "required": false
```
Comment (sub-mod) [inline tensorflow-build-job.json]: give a default value of  ""

File: tensorflow-build-job.json [hunk 2]
```diff
@@ -175,11 +189,41 @@
...
 						"restartPolicy": "Never"
                     }
                 }
             }
+        },
+        {
+          "apiVersion": "v1",
+          "kind": "Secret",
+          "metadata": {
+            "name": "tensorflow-secret",
```
Comment (sub-mod) [inline tensorflow-build-job.json]: call the Secret name  as  ${APPLICATION_NAME}-secret

File: tensorflow-build-job.json [hunk 3]
```diff
@@ -175,11 +189,41 @@
                                     }
                                 }
                             }
+                        ],
+                        "volumes": [
+                            {
+                              "name": "ssh-config",
+                              "secret": {
+                                "secretName": "tensorflow-secret",
```
Comment (sub-mod) [inline tensorflow-build-job.json]: call the Secret name as ${APPLICATION_NAME}-secret

File: tensorflow-build-job.json [hunk 4]
```diff
@@ -157,12 +157,26 @@
...
-                                        "value": "${GIT_DEST_REPO}"
+                                        "name": "GIT_RELEASE_REPO",
+                                        "value": "${GIT_RELEASE_REPO}"
+                                    },
+                                    {
+                                        "name": "PAGURE_COMMIT_REPO",
+                                        "value": "${PAGURE_COMMIT_REPO}"
+                                    },
+                                    {
+                                        "name": "PAGURE_COMMIT_WHL",
```
Comment (sub-mod) [inline tensorflow-build-job.json]: maybe,  rename to USE_PAGURE_COMMIT

File: tensorflow-build-dc.json [hunk 5]
```diff
@@ -495,10 +539,27 @@
...
+            "value": "ssh://git@pagure.io/tensorflow.git",
+            "required": true
+        },
+        {
+            "name": "PAGURE_SSH_PRIVATE_KEY",
+            "description": "SSH private key to authenticate the public key set in GitHub.",
+            "required": false
+        },
+        {
+            "name": "PAGURE_COMMIT_WHL",
```
Comment (sub-mod) [inline tensorflow-build-dc.json]: maybe rename as USE_PAGURE_COMMIT

File: tensorflow-build-dc.json [hunk 6]
```diff
@@ -268,6 +297,21 @@
                     "targetPort": "8080-tcp"
                 }
             }
+        },
+        {
+          "apiVersion": "v1",
+          "kind": "Secret",
+          "metadata": {
+            "name": "tensorflow-secret",
```
Comment (sub-mod) [inline tensorflow-build-dc.json]: rename to ${APPLICATION_NAME}-secret

File: tensorflow-build-dc.json [hunk 7]
```diff
@@ -495,10 +539,27 @@
...
+        {
+            "name": "PAGURE_COMMIT_REPO",
+            "description": "PAGURE_COMMIT_REPO",
+            "value": "ssh://git@pagure.io/tensorflow.git",
+            "required": true
+        },
+        {
+            "name": "PAGURE_SSH_PRIVATE_KEY",
+            "description": "SSH private key to authenticate the public key set in GitHub.",
+            "required": false
```
Comment (sub-mod) [inline tensorflow-build-dc.json]: give a default value of ""

File: tensorflow-build-dc.json [hunk 8]
```diff
@@ -218,6 +232,21 @@
                                     }
                                 }
                             }
+                        ],
+                        "volumes": [
+                            {
+                              "name": "ssh-config",
+                              "secret": {
+                                "secretName": "tensorflow-secret",
```
Comment (sub-mod) [inline tensorflow-build-dc.json]: rename to ${APPLICATION_NAME}-secret

File: s2i/bin/run [hunk 9]
```diff
@@ -385,16 +387,22 @@ else
...
 						cp /workspace/bins/build_info.yaml /tmp/release_dir/
 						cp /workspace/bins/*.whl /tmp/release_dir/
-						ls -l
 						whlff=$(basename `ls *.whl`)
 						OSVER=$(. /etc/os-release;echo $ID$VERSION_ID)
 						#OSVER=$(cat build_info.yaml | grep -e "OS_VER" | awk '{print $2,$4}' |tr -d " ")
 						RELEASE_NAME=$OSVER/$NB_PYTHON_VER/$TF_GIT_BRANCH
-						cd wheels
-						ls -l
+						for varname in ${!TF_NEED_*}; do
```
Comment (sub-mod) [inline s2i/bin/run]: This logic can goto pagure_commit.sh

File: tensorflow-build-dc.json [hunk 10]
```diff
@@ -268,6 +297,21 @@
...
                 }
             }
+        },
+        {
+          "apiVersion": "v1",
+          "kind": "Secret",
+          "metadata": {
+            "name": "tensorflow-secret",
+            "labels": {
+                "appTypes": "tensorflow-build-job",
```
Comment (sub-mod) [inline tensorflow-build-dc.json]: should be tensorflow-build-dc

**Review by sub-mod — CHANGES_REQUESTED** [review]

**Comment by sub-mod** [comment]
LGTM 
Thanks.

File: build_tools/pagure_commit.sh [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+echo "=============================="
+echo "TENSORFLOW_BUILD_DIR_NAME="$TENSORFLOW_BUILD_DIR_NAME
+echo "NOTES="$GIT_COMMIT_MSG
+echo "FILES="$FILES
+echo "BRANCH="$BRANCH
+echo "=============================="
+
+
+# Config: Script commit files on behalf of
+git config --global user.name "Harshad Reddy Nalla"
```
Comment (goern) [inline build_tools/pagure_commit.sh:28]: please set the user name to "Red Hat's AICoE"

File: build_tools/pagure_commit.sh [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+git config --global user.email "hnalla@redhat.com"
```
Comment (goern) [inline build_tools/pagure_commit.sh:29]: please set the email to "goern+aicoe@redhat.com" ;)

**Review by goern — COMMENTED** [review]
uh, here are my comments...

File: build_tools/pagure_commit.sh [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+	sed -i "/<section>/a $TAG" index.html
+}
+
+
+PUSH_DATA(){
+	for f in $FILES;do
+		echo "FILE="$f
+		cp $f $TENSORFLOW_BUILD_DIR_NAME/
+	done
+	CHECK_LOCAL_AND_UPSTREAM
```
Comment (goern) [inline build_tools/pagure_commit.sh:64]: uh ah, we need to generate python simple index files... just like https://pypi.org/simple/tensorflow/

**Review by goern — COMMENTED** [review]
uh ah, we need to generate python simple index files... just like https://pypi.org/simple/tensorflow/

File: build_tools/pagure_commit.sh (hunk 13 above)
Comment (fridex) [reply build_tools/pagure_commit.sh:64]: I'm just wondering... can it generate web server for us? :)

File: build_tools/pagure_commit.sh (hunk 12 above)
Comment (harshad16) [reply build_tools/pagure_commit.sh:29]: Please add and give the account user commit permissions on pagure/tensorflow.
@goern

File: build_tools/pagure_commit.sh (hunk 13 above)
Comment (harshad16) [reply build_tools/pagure_commit.sh:64]: We want all the wheel files to be listed on a simple HTTP server?

Comment (fridex) [reply build_tools/pagure_commit.sh:64]: It's for pipenv/pip not for a human so we need to do it in pipenv/pip consumable way. I would say let's check whether a simple web server listing is sufficient there.

**Pull request merged**